draft-ietf-tls-esni.md

```diff
@@ -252,34 +257,44 @@ specification, the contents are an `ECHConfigContents` structure.
 
 The `ECHConfigContents` structure contains the following fields:
 
+key_config
+: A `HpkeKeyConfig` structure carrying the configuration information associated
+with the HPKE public key.
+
+maximum_name_length
+: The longest name of a backend server, if known. If this value is not known it
+can be set to zero, in which case clients SHOULD use the inner ClientHello
+padding scheme described below. That could happen if wildcard names are in use,
+or if names can be added or removed from the anonymity set during the lifetime
+of a particular ECH configuration.
+
 public_name
 : The non-empty name of the client-facing server, i.e., the entity trusted to
 update the ECH configuration. This is used to correct misconfigured clients, as
 described in {{handle-server-response}}.
 
-public_key
-: The HPKE public key used by the client to encrypt ClientHelloInner.
+extensions
+: A list of extensions that the client must take into consideration when
+generating a ClientHello message. These are described below
+({{config-extensions}}).
+
+The `HpkeKeyConfig` structure contains the following fields:
+
+key_id
```

Comment:
Why `key_id` instead of `config_id`? The new name sounds as if it should be possible to have a single key used by multiple configs, but what we actually want is a handle with which the server can resolve which config to use.

Reply:
Since the identifier corresponds primarily to the HPKE public key, and not the public name et al. (If we lifted the key_id *out* of the HpkeKeyConfig, I think config_id would be better.)

Reply:
Perhaps another way to resolve this would be to change `key_id` to simply `id`. The `key_` prefix seems a bit redundant.

Reply:
We could do that, but then ClientECH would either have an inconsistently named field (key_id) or an ambiguously named field (id). I'd prefer to keep key_id.